crowbar_framework/app/models/service_object.rb

```diff
@@ -208,23 +208,21 @@ def reset_proposal(inst, bc = @bc_name)
         end
       end
 
-      nodes = NodeObject.find("roles:#{bc}-config-#{inst}").map do |node|
-        if node.role.nil?
-          false
-        else
-          node.role.tap do |role|
-            return true if role.default_attributes["state"] == "ready"
-
-            role.default_attributes["state"] = "ready"
-            role.save
-          end
+      nodes = []
+      NodeObject.find("roles:#{bc}-config-#{inst}").each do |node|
+        nodes.push(node.alias) unless node.role
+
+        next if node.role.default_attributes["state"] == "ready"
+        node.role.default_attributes["state"] = "ready"
```

Comment:
Doesn't matter much, but you could simply use `node.crowbar["state"]` instead of `node.role.default_attributes["state"]`.